spec/std/wait_group_spec.cr

```diff
@@ -0,0 +1,104 @@
+require "spec"
+require "wait_group"
+
+describe WaitGroup do
+  describe "add" do
+    it "can't decrement to a negative counter" do
+      wg = WaitGroup.new
+      wg.add(5)
+      wg.add(-3)
+      expect_raises(Exception) { wg.add(-5) }
+    end
+  end
+
+  describe "done" do
+    it "can't decrement to negative value" do
+      wg = WaitGroup.new
+      wg.add(1)
+      wg.done
+      expect_raises(Exception) { wg.done }
+    end
+  end
+
+  it "waits until concurrent executions are done" do
+    wg1 = WaitGroup.new
+    wg2 = WaitGroup.new
+
+    8.times do
+      wg1.add(16)
+      wg2.add(16)
+      exited = Channel(Bool).new(16)
+
+      16.times do
+        spawn do
+          wg1.done
+          wg2.wait
+          exited.send(true)
+        end
+      end
+
+      wg1.wait
+
+      16.times do
+        select
+        when exited.receive
+          raise "WaitGroup released group too soon"
```

Comment:
In Minitest I would call `flunk message` to fail a test. Is there an equivalent in spec?

Reply:
[`fail`](https://crystal-lang.org/api/master/Spec/Methods.html#fail%28msg%2Cfile%3D__FILE__%2Cline%3D__LINE__%29-instance-method)